pandas/tests/io/test_excel.py

```diff
@@ -1597,53 +1463,44 @@ def test_to_excel_multiindex_no_write_index(self):
         multi_index = MultiIndex.from_tuples([(70, 80), (90, 100)])
         frame2.index = multi_index
 
-        with ensure_clean(self.ext) as path:
-
-            # Write out to Excel without the index.
-            frame2.to_excel(path, 'test1', index=False)
-
-            # Read it back in.
-            reader = ExcelFile(path)
-            frame3 = read_excel(reader, 'test1')
+        # Write out to Excel without the index.
+        frame2.to_excel(self.path, 'test1', index=False)
 
-            # Test that it is the same as the initial frame.
-            tm.assert_frame_equal(frame1, frame3)
+        # Read it back in.
+        reader = ExcelFile(self.path)
+        frame3 = read_excel(reader, 'test1')
 
-    def test_to_excel_float_format(self):
-        _skip_if_no_xlrd()
+        # Test that it is the same as the initial frame.
+        tm.assert_frame_equal(frame1, frame3)
 
+    def test_to_excel_float_format(self, merge_cells, engine, ext):
         df = DataFrame([[0.123456, 0.234567, 0.567567],
                         [12.32112, 123123.2, 321321.2]],
                        index=['A', 'B'], columns=['X', 'Y', 'Z'])
 
-        with ensure_clean(self.ext) as filename:
-            df.to_excel(filename, 'test1', float_format='%.2f')
-
-            reader = ExcelFile(filename)
-            rs = read_excel(reader, 'test1', index_col=None)
-            xp = DataFrame([[0.12, 0.23, 0.57],
-                            [12.32, 123123.20, 321321.20]],
-                           index=['A', 'B'], columns=['X', 'Y', 'Z'])
-            tm.assert_frame_equal(rs, xp)
+        df.to_excel(self.path, 'test1', float_format='%.2f')
 
-    def test_to_excel_output_encoding(self):
-        _skip_if_no_xlrd()
+        reader = ExcelFile(self.path)
+        rs = read_excel(reader, 'test1', index_col=None)
+        xp = DataFrame([[0.12, 0.23, 0.57],
+                        [12.32, 123123.20, 321321.20]],
+                       index=['A', 'B'], columns=['X', 'Y', 'Z'])
+        tm.assert_frame_equal(rs, xp)
 
+    def test_to_excel_output_encoding(self, merge_cells, engine, ext):
         # avoid mixed inferred_type
         df = DataFrame([[u'\u0192', u'\u0193', u'\u0194'],
                         [u'\u0195', u'\u0196', u'\u0197']],
                        index=[u'A\u0192', u'B'],
                        columns=[u'X\u0193', u'Y', u'Z'])
 
-        with ensure_clean('__tmp_to_excel_float_format__.' + self.ext)\
-                as filename:
+        with ensure_clean('__tmp_to_excel_float_format__.' + ext) as filename:
```

Comment:
This and the next test modified the name of the file generated by `ensure_clean`. In these tests, the content manager within the setup fixture is therefore a no-op. Could move these into a separate class or do something tricky to prevent that, but I figure it was easiest just to keep as is